aicsdeformation/preprocessors/PIVbatcher.py

```diff
@@ -0,0 +1,534 @@
+import sys
+
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+print(plt.get_backend())
+
+import openpiv.tools
+import openpiv.process
+import openpiv.scaling
+import openpiv.validation
+import openpiv.filters
+import numpy as np
+import pandas as pd
+
+import os
+from pathos.multiprocessing import ProcessingPool as Pool
+from functools import partial
+
+import glob
+
+print(plt.get_backend())
+
+
+# Class from Forrest for doing multi-threaded processing; used in calculate_displacements function to speed up
+# processing across multiple cores.
+class WithPool(Pool):
+    """pathos ProcessingPool with functioning __exit__ call
+    Parameters
+    ----------
+    *args
+        variable length argument list matching input
+        to pathos.multiprocessing.Pool
+    **kwargs
+        keyword argument input matching pathos.multiprocessing.Pool
+    Examples
+    --------
+    >>> with WithPool(number_processes) as pool:
+    >>>     pool.map(myfunc, myInput)
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(WithPool, self).__init__(*args, **kwargs)
+
+    def __exit__(self, *args, **kwargs):
+        super(WithPool, self)._clear()
+
+
+class PIVbatcher(object):
+
+    def __init__(self, path, window, overlap, search_area, opath):
+        """
+        Set up the control member variables needed by PIVbatcher.
+
+        Example
+        -------
+        PIVbatcher(
+                '/AssayDev/MicroscopyOtherData/Melissa/20180116_PIV_slides/20171220_PIV_data_round2',
```

Comment:
A variable rather than a hardcoded path is preferable so thanks for that. What strategy do you prefer? Parameters, a params.json file, and / or running from a jupyter instance?